### DISCUSSION TITLE
Setting up Dual mining ETH+TON in Teamredminer

### DISCUSSION THREAD

**Discussion by Celldfusion** [RainbowMiner/RainbowMiner#2067] [Q&A] [answered]
Hello, I am trying to setup dual mining in eth+ton on teamredminer but it still just use ethash. I already setup a ton wallet, ton pool (icemining) and i already configured the miners.config file to include sha256ton as secondary algorithm, but it still doesnt work. I dont know what i am missing or what to do.

**Comment by RainbowMiner** [comment]
Could you please create a Debug file and upload it here (click "Debug file" on page http://localhost:4000, left-hand side)? It's maybe a simple configuration thing.

**Reply by Celldfusion** [reply]
[debug_2022-05-11.zip](https://github.com/RainbowMiner/RainbowMiner/files/8667884/debug_2022-05-11.zip)
here is the debug file

**Reply by RainbowMiner** [reply]
Ok, thank you for upload that file. Thanks to that, I could isolate a small bug - we are missing an entry in our algorithms database.
I'll quickly push a new follow-up release, so that you can do an autoupdate on your RainbowMiner instance.

Also, empty the FocusWallet parameters in pools.config.txt in sections Icemining and Flexpool `"FocusWallet": "",` - this setting can interfere with dual-mining.

**Reply by Celldfusion** [reply]
So its a bug, thank you. I will wait for the update and check if the problem will be solved

**Reply by RainbowMiner** [reply]
Ok, I have pushed a new release, that contains the bugfix. Please make sure to empty your "Params" parameter in miners.config.txt before testing the new releases, otherwise you might have some parameter-clashes.

**Reply by Celldfusion** [reply]
It still  doesnt work. Now it shows me that error.
![image](https://user-images.githubusercontent.com/105326702/167897832-797c0600-cbb6-43b2-8efa-7cbbeabcc2e3.png)

**Reply by RainbowMiner** [reply]
Ohhh! I see. Sorry for that. Since I have no AMD GPUs in my rigs, I sometimes miss a bug. I'll quickly fix it, then you can simply download the Teamred miner module.

**Reply by Celldfusion** [reply]
Oh i understand, no problem, i will wait for the fix :)

**Reply by RainbowMiner** [reply]
In the meantime, clear the "Intensity" setting in miners.config.txt and use "Params" to set "--dual_intensity=..." in there.

**Reply by Celldfusion** [reply]
Okay, intensity line cleared

**Reply by RainbowMiner** [reply]
The commit 8bd1d58e80ae5958f141a22a1ee8a93ae94d3826 should solve the error. Now you can run benchmarks for multiple intensities, like it was intended. 
For example, if you set the parameter `"Intensity": "0.25,0.5,0.75",` RainbowMiner would run three benchmarks with the specific dual mining algorithms with the three different intensities 0.25, 0.5 and 0.75. The Teamred miner allows setting a value between 0.0 and 1.0 (see their USAGE.txt).

**Reply by Celldfusion** [reply]
It solved the problem with the red letters in the image above, but the problem still persists, sadly. Even if i select specific algos (including sha256 and ethash),selecting all algos, or editing the miners.config file, Rainbowminer doesnt run benchmark por dual mining in my gpu (ethash+256). Its like it doesnt exist for the program, or at least doesnt exist for amd gpus. And i confirmed that the algo sha256ton is added correctly because lolminer benchmark it, but just sha256ton alone, and i want to dual mining on teamredminer. I dont know if i am doing something wrong or its a program issue :c

**Comment by Celldfusion** [comment, marked as answer]
Issue solved in [#2069](https://github.com/RainbowMiner/RainbowMiner/issues/2069)
